src/dockerflow/fastapi/checks.py

```diff
@@ -0,0 +1,52 @@
+import functools
+import logging
+from dataclasses import dataclass
+from typing import Dict, List, Tuple
+
+from ..checks import level_to_text
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class CheckDetail:
+    status: str
+    level: int
+    messages: Dict[int, str]
+
+
+registered_checks = dict()
+
+
+def register_heartbeat_check(func=None, *, name=None):
+    if func is None:
+        return functools.partial(register_heartbeat_check, name=name)
+
+    if name is None:
+        name = func.__name__
+
+    logger.debug("Registered Dockerflow check %s", name)
+
+    @functools.wraps(func)
+    def decorated_function(*args, **kwargs):
+        logger.debug("Called Dockerflow check %s", name)
+        return func(*args, **kwargs)
+
+    registered_checks[name] = decorated_function
+    return decorated_function
+
+
+def _heartbeat_check_detail(check):
+    errors = check()
+    level = max([0] + [e.level for e in errors])
+    return CheckDetail(
+        status=level_to_text(level), level=level, messages={e.id: e.msg for e in errors}
```

Comment:
This bit worried me - it looks like if two errors have the same level, just one will win:

`messages={e.id: e.msg for e in errors}`

However, the other platforms seem to work that way as well, so 🤷